web/src/compose_recipient.js

```diff
@@ -100,13 +103,44 @@ export function open_compose_stream_dropup() {
     $("#id_compose_select_recipient > .dropdown-toggle").trigger("click");
 }
 
+function switch_message_type(message_type) {
+    $("#compose-content .alert").hide();
+
+    compose_state.set_message_type(message_type);
+
+    const opts = {
+        message_type,
+        trigger: "toggle message",
+        stream: compose_state.stream_name(),
+        topic: compose_state.topic(),
+        private_message_recipient: compose_state.private_message_recipient(),
+    };
+    compose_actions.show_compose_box(message_type, opts);
+}
+
 export function on_compose_select_recipient_update(new_value) {
-    const $stream_header_colorblock = $("#compose_recipient_selection_dropdown").find(
-        ".stream_header_colorblock",
-    );
-    stream_bar.decorate(new_value, $stream_header_colorblock);
+    const message_type = compose_state.get_message_type();
+    if (new_value === PRIVATE_MESSAGE) {
+        if (message_type !== "private") {
+            switch_message_type("private");
+        }
+        if (compose_state.private_message_recipient().length === 0) {
+            $("#private_message_recipient").trigger("focus").trigger("select");
+        }
+    } else {
+        const $stream_header_colorblock = $("#compose_recipient_selection_dropdown").find(
+            ".stream_header_colorblock",
+        );
```

Comment:
Is there a reason this uses .find, rather than just a compound selector (`$("#compose_recipient_selection_dropdown .stream_header_colorblock")`)?

Reply:
I guess this is just how it was before -- so the answer is probably no. Might be worth cleaning up but in its own commit at the end might be better.

Reply:
Yeah I just used it as I found it. I can add a clean up commit for that.